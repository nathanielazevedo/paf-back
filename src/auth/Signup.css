
.signup-container{
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: #1c2227;
}

.signup-form{
  display: flex;
  flex-direction: column;
  padding: 50px;
  background-color: #121212;
}

.signup-title{
  color: white;
}

.signup-input{
  background-color: transparent;
  border: none;
  border-bottom: solid white 1px;
  margin: 20px 0;
  color: white;
}