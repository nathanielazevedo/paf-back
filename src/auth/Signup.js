/** @format */

import React from "react";
import "./Signup.css";
import Form from "../form/Form";

//SignupFunc is passed down from App.js as a prop. This component passes the signupFunc down to the form.

function Signup({ signupFunc }) {
  return (
    <div className="signup-container">
      <Form
        title="Sign Up"
        inputs={["username", "password", "email", "firstName", "lastName"]}
        func={signupFunc}
        after="/paf-front-end/friends"
      />
    </div>
  );
}

export default Signup;
