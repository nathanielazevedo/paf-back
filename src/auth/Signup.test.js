/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Signup from "./Signup";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Signup />
      </UserProvider>
    </MemoryRouter>
  );
});


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Signup />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays expected text", function () {
  const { getAllByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Signup />
      </UserProvider>
    </MemoryRouter>
  );

  expect(getAllByText("Sign Up")[0]).toBeInTheDocument();
});   