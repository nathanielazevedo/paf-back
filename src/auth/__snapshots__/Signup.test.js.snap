// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="signup-container"
  >
    <div
      class="form-container"
    >
      <div
        class="form-inner"
      >
        <h3
          class="form-title"
        >
          Sign Up
        </h3>
        <form
          class="form-input-container"
        >
          <div
            class="form-input-group"
          >
            <label
              for="username"
            >
              username
            </label>
            <input
              class="form-input"
              data-testid="username"
              name="username"
              placeholder="username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-input-group"
          >
            <label
              for="password"
            >
              password
            </label>
            <input
              class="form-input"
              data-testid="password"
              name="password"
              placeholder="password"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-input-group"
          >
            <label
              for="email"
            >
              email
            </label>
            <input
              class="form-input"
              data-testid="email"
              name="email"
              placeholder="email"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-input-group"
          >
            <label
              for="firstName"
            >
              firstName
            </label>
            <input
              class="form-input"
              data-testid="firstName"
              name="firstName"
              placeholder="firstName"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-input-group"
          >
            <label
              for="lastName"
            >
              lastName
            </label>
            <input
              class="form-input"
              data-testid="lastName"
              name="lastName"
              placeholder="lastName"
              required=""
              type="text"
              value=""
            />
          </div>
          <button
            class="form-button"
            data-testid="button"
            type="submit"
          >
            Sign Up
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
